manager/src/main/java/com/redhat/service/bridge/manager/connectors/ConnectorAuthImpl.java

```diff
@@ -0,0 +1,94 @@
+package com.redhat.service.bridge.manager.connectors;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URLEncoder;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.nio.charset.StandardCharsets;
+import java.time.Duration;
+
+import javax.enterprise.context.RequestScoped;
+
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+
+import io.vertx.core.json.JsonObject;
+
+@RequestScoped
+public class ConnectorAuthImpl implements ConnectorAuth {
+
+    @ConfigProperty(name = "mc.auth.serverUrl")
+    String authServerUrl;
+
+    @ConfigProperty(name = "mc.auth.clientId")
+    String clientId;
+
+    @ConfigProperty(name = "mc.auth.tokenPath")
+    String tokenPath;
+
+    @ConfigProperty(name = "mc.auth.offlineToken")
+    String offlineToken;
+
+    String bearerToken;
+
+    // TODO-MC this should be cached for five minutes
+    @Override
+    public String bearerToken() {
+        if (bearerToken == null) {
+            bearerToken = bearerTokenFromOfflineToken(offlineToken);
+        }
+        return null;
+    }
+
+    /**
+     * Copied from https://github.com/redhat-developer/app-services-operator/blob/main/source/rhoas/src/main/java/com/openshift/cloud/beans/AccessTokenSecretTool.java#L117
+     * This method exchanges an offline token for a new refresh token
+     *
+     * @param offlineToken the token from ss.redhat.com
+     * @return a token to be used as a bearer token to authorize the user
+     */
+    private String bearerTokenFromOfflineToken(String offlineToken) {
+        try {
+            HttpRequest request =
+                    HttpRequest.newBuilder().uri(URI.create(authServerUrl + "/" + tokenPath))
+                            .header("content-type", "application/x-www-form-urlencoded")
+                            .timeout(Duration.ofMinutes(2)).POST(ofFormData("grant_type", "refresh_token",
+                                    "client_id", clientId, "refresh_token", offlineToken))
+                            .build();
+
+            HttpClient client = HttpClient.newBuilder().build();
```

Comment:
Just wondering if it makes sense to rely on the VertX HTTP client here? We already use that HTTP Client in the Shard, so it would be good to keep our approach to HTTP requests consistent across the project.

Reply:
Will use the Quarkus OID https://issues.redhat.com/browse/MGDOBR-196